crates/install-wheel-rs/src/linker.rs

```diff
@@ -302,7 +302,7 @@ fn clone_recursive(site_packages: &Path, wheel: &Path, entry: &DirEntry) -> Resu
     debug!("Cloning {} to {}", from.display(), to.display());
 
     // Attempt to copy the file or directory
-    let reflink = reflink_copy::reflink(&from, &to);
+    let reflink = reflink_copy::reflink_or_copy(&from, &to);
```

Comment:
I think we need our own version of this that behaves differently when the error is `std::io::ErrorKind::AlreadyExists`.

Reply:
In that case, copying isn't the right fallback, I think?

Reply:
Hence the test failure. Notice that we have a branch for `AlreadyExists` below, where we recurse.

Reply:
Oh, it looks like `reflink_or_copy` also doesn't fallback properly when the target is a directory.

Reply:
We need a `reflink_or_copy_all`, but if we need to handle `AlreadyExists` with a no-fallback path we probably can't do it upstream in the `reflink_copy` crate.

Reply:
Yeah, we should just write our own here.

Reply:
I could contribute an upstream PR to add a `reflink_or_copy_all` function.